### PR TITLE
Component/vf smooth scroll

### PR DESCRIPTION
**Pull request by khawkins98**
A long-time to do list item of adding a `.vf-smooth-scroll`; two notes:

1. Is opt-in by adding the component+class, as not all scenarios benefit from smooth scrolling
2. Has some extra CSS for `prefers-reduce-motion` and to not smooth scroll the browser in-page search

https://user-images.githubusercontent.com/928100/141148349-60365de8-b038-4cba-8113-586a045eb682.mov

### CODE REVIEW

File: components/vf-smooth-scroll/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## About
+
+This enables smooth scrolling when a user clicks on anchor links or other in-page scrolling is triggered.
+
+## Usage
+
+This is an opt-in behaviour that is activated by putting the `vf-smooth-scroll` class on a parent element — a good place would be at `<html class="vf-smooth-scroll">` (note: it probably won't work if you add it your `<body>` element).
+
+Not that as of November 2021, CSS-based smooth scrolling [is still not support by Safari on Mac OS or iOS](https://caniuse.com/css-scroll-behavior).
```
Comment (nitinja) [inline components/vf-smooth-scroll/README.md:13]: Seems to be working on safari desktop https://stackoverflow.com/questions/19012495/smooth-scroll-to-div-id-jquery/61918221?noredirect=1#comment123675898_61918221

File: components/vf-smooth-scroll/vf-smooth-scroll.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// when the user clicks on a page (doesn't use native search)
+// More at https://schepp.dev/posts/smooth-scrolling-and-page-search/
+@keyframes vfsmoothscroll1 {
+  from, to { scroll-behavior: smooth; }
+}
+@keyframes vfsmoothscroll2 {
+  from, to { scroll-behavior: smooth; }
+}
+.vf-smooth-scroll:focus-within {
+  animation-name: vfsmoothscroll2;
```
Comment (nitinja) [inline components/vf-smooth-scroll/vf-smooth-scroll.scss:28]: nonblocking: Can same animation work for both cases instead of two identical ones?

**Review by nitinja — APPROVED** [review]

File: components/vf-smooth-scroll/README.md (hunk 1 above)
Comment (khawkins98) [reply components/vf-smooth-scroll/README.md:13]: I know Safari desktop's technical preview supports it, perhaps it's that?

File: components/vf-smooth-scroll/vf-smooth-scroll.scss (hunk 2 above)
Comment (khawkins98) [reply components/vf-smooth-scroll/vf-smooth-scroll.scss:28]: It seems to be required, the two animation frames allows the focus handoff between the DOM and browser UI to be more graceful. 

I've tried consolidating them and the smooth scroll no longer functions reliably.

It's a bit of a hack that may no longer be needed in a year or two.

File: components/vf-smooth-scroll/README.md (hunk 1 above)
Comment (nitinja) [reply components/vf-smooth-scroll/README.md:13]: not sure, My assumption is based on SO comment

**Pull request merged**